crates/ika-move-packages/packages/ika_system/sources/dwallet/dwallet_2pc_mpc_coordinator_inner.move

```diff
@@ -1131,24 +1179,29 @@ public(package) fun create_message_approval(
 ///
 /// ### Aborts
 /// - Aborts if the provided `hash_scheme` is not supported by the system (checked during `create_message_approval`).
-public fun approve_message(
+public(package) fun approve_message(
+    self: &DWalletCoordinatorInner,
     dwallet_cap: &DWalletCap,
+    signature_algorithm: u8,
     hash_scheme: u8,
     message: vector<u8>
 ): MessageApproval {
-    create_message_approval(
-        dwallet_cap.dwallet_id,
+    let dwallet_id = dwallet_cap.dwallet_id;
+    let (dwallet, _) = self.get_active_dwallet_and_public_output(dwallet_id);
+    assert!(self.supported_curves_to_signature_algorithms.contains(&dwallet.curve), EInvalidCurve);
+    let supported_curve_to_signature_algorithms = self.supported_curves_to_signature_algorithms[&dwallet.curve];
+    assert!(supported_curve_to_signature_algorithms.contains(&signature_algorithm), EInvalidSignatureAlgorithm);
+    assert!(self.supported_signature_algorithms_to_hash_schemes.contains(&signature_algorithm), EInvalidSignatureAlgorithm);
```

Comment:
I think its much better to check all of these together rather than separately. I could foresee a reason why not to use specific hash in a sepcific signature algorithm for one curve but not the other.

Let's do that you just have an array of supported combinations, and you just put there all the supported combinations and check if the current combination exists, i.e. (curve, signature_scheme, hash_scheme)